Source/Diagnostics/Diagnostics.H

```diff
@@ -22,7 +22,26 @@
 #include <vector>
 
 /** All types of diagnostics. */
-enum struct DiagTypes {Full, BackTransformed, BoundaryScraping, TimeAveraged};
+enum struct DiagTypes
+{
+    Full,
+    BackTransformed,
+    BoundaryScraping,
+    TimeAveraged
+};
+
+/** A struct containing the parameters required to initialize a diagnostics */
+struct InitDiagnosticsParameters
+{
+    int finest_level;
+    int max_level;
+
+    int do_moving_window;
+    int moving_window_dir;
+    amrex::Real moving_window_x;
+    amrex::Real cell_size_lev0_mwdir;
```

Comment:
As a purist, one could make them `std::optional` to ensure no UB can occur :nerd_face: